examples/twentyseven-pt-stencil-solver/twentyseven-pt-stencil-solver.cpp

```diff
@@ -0,0 +1,365 @@
+            const auto index_left = dp * j + dp * dp * k;
+            const auto index_right = dp * j + dp * dp * k + (dp - 1);
+
+            for (int b = -1; b <= 1; ++b) {
+                for (int c = -1; c <= 1; ++c) {
+                    rhs[index_left] -= u(0.0, yi + b * h, zi + c * h) *
+                                       coefs[3 * (b + 1) + 3 * 3 * (c + 1)];
+                    rhs[index_right] -=
+                        u(1.0, yi + b * h, zi + c * h) *
+                        coefs[3 * (b + 1) + 3 * 3 * (c + 1) + 2];
+                }
+            }
+        }
+    }
+
+    // To avoid double counting we have to check if our previous calculations
+    // included this case
+    for (size_t i = 0; i < dp; ++i) {
+        for (size_t k = 0; k < dp; ++k) {
+            const auto xi = (i + 1) * h;
+            const auto zi = (k + 1) * h;
+            const auto index_top = i + dp * dp * k;
+            const auto index_bot = i + dp * dp * k + dp * (dp - 1);
+
+            for (int a = -1; a <= 1; ++a) {
+                if ((i < (dp - 1) || a < 1) && (i > 0 || a > -1)) {
+                    for (int c = -1; c <= 1; ++c) {
+                        rhs[index_top] -= u(xi + a * h, 0.0, zi + c * h) *
+                                          coefs[(a + 1) + 3 * 3 * (c + 1)];
+                        rhs[index_bot] -=
+                            u(xi + a * h, 1.0, zi + c * h) *
+                            coefs[(a + 1) + 3 * 3 * (c + 1) + 3 * 2];
+                    }
+                }
+            }
+        }
+    }
+
+    // Now every side has to be checked
+    for (size_t i = 0; i < dp; ++i) {
+        for (size_t j = 0; j < dp; ++j) {
+            const auto xi = (i + 1) * h;
+            const auto yi = (j + 1) * h;
+            const auto index_front = i + dp * j;
+            const auto index_back = i + dp * j + dp * dp * (dp - 1);
+
+            for (int a = -1; a <= 1; ++a) {
+                if ((i < (dp - 1) || a < 1) && (i > 0 || a > -1)) {
+                    for (int b = -1; b <= 1; ++b) {
+                        if ((j < (dp - 1) || b < 1) && (j > 0 || j > -1)) {
+                            rhs[index_front] -= u(xi + a * h, yi + b * h, 0.0) *
+                                                coefs[(a + 1) + 3 * (b + 1)];
+                            rhs[index_back] -=
+                                u(xi + a * h, yi + b * h, 1.0) *
+                                coefs[(a + 1) + 3 * (b + 1) + 3 * 3 * 2];
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+
+// Prints the solution `u`.
+void print_solution(int dp, const double *u)
+{
+    for (size_t k = 0; k < dp; ++k) {
+        for (size_t j = 0; j < dp; ++j) {
+            for (size_t i = 0; i < dp; ++i) {
+                std::cout << u[i + dp * (j + dp * k)] << ' ';
+            }
+            std::cout << std::endl;
+        }
+        std::cout << ':' << std::endl;
```

Comment:
Same as previously: You can replace both above `std::endl` with `'\n'` if you want slightly more performance while printing. The buffer will be flushed with the `std::endl` below.

Reply:
Just pushed the update. Normally the amount of output isn't too big so I didn't bother to use simple linebreaks.

Reply:
I just wanted to mention it, and I don't know how big the output normally is.